supervision/video/backends/base.py

```diff
@@ -0,0 +1,118 @@
+from __future__ import annotations
+
+from collections.abc import Iterator
+from typing import Protocol, runtime_checkable
+
+import numpy as np
+
+from ..utils import VideoInfo
+
+
+@runtime_checkable
+class Backend(Protocol):
+    """
+    The high-level :pyclass:`~supervision.video.Video` adapter instantiates a
+    backend - selected by name - and then only calls the methods defined
+    below.  Anything else is considered a private implementation detail.
+    """
+
+    def __init__(self, source: str | int):
+        """Create a new backend for source.
+
+        ``source`` can be
+        * ``str`` - file path, RTSP/HTTP URL …
+        * ``int`` - webcam index (OpenCV-style)
+        """
```

Comment:
All docstrings must follow the Google style format compatible with MKDocs. Ensure every docstring is updated to meet these guidelines.